clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -3075,6 +3075,55 @@ LogicalResult GetRuntimeMemberOp::verify() {
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// GetMethodCalleeOp Definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult GetMethodCalleeOp::verify() {
+  auto methodTy = getMethod().getType();
+  auto objectPtrTy = mlir::cast<mlir::cir::PointerType>(getObject().getType());
+  auto objectTy = objectPtrTy.getPointee();
+
+  if (methodTy.getClsTy() != objectTy) {
+    emitError() << "method class type and object type do not match";
+    return mlir::failure();
+  }
+
+  auto calleePtrTy = mlir::cast<mlir::cir::PointerType>(getCallee().getType());
+  auto calleeTy = mlir::cast<mlir::cir::FuncType>(calleePtrTy.getPointee());
+  auto methodFuncTy = methodTy.getMemberFuncTy();
+
+  // Assume:
+  // - objectTy is !cir.ptr<!T>
+  // - methodFuncTy is !cir.func<!Ret (!Args)>
+  //
+  // We verify at here that:
+  // - calleeTy is !cir.func<!Ret (!cir.ptr<!T>, !Args)>
+  if (methodFuncTy.getReturnType() != calleeTy.getReturnType()) {
```

Comment:
Is this possible to constrain this in tablegen?

Reply:
I've looked at the builtin traits but I can't find a way to constraint this. Is there anything other than the special traits that can help here?